### ISSUE TITLE
disk writing

### ISSUE THREAD

**Issue by chaeyoungdul** [vincentneo/LosslessSwitcher#28]
hi,
I used the app for a couple of times, but I noticed it was writing and reading the disk way more than it needs or should.
I saw it read 55 gigabytes and write 11 in 2 hours.

**Comment by vincentneo** [comment]
Very interesting, how did you see the read/write metrics? 
Is it incurred on storage or memory, and is it just a metrics for how much is read/written rather than persisted amount?

It doesn't surprise me that it reads so much, since it reads `OSLogStore` (like the Console app), fairly often, to monitor latest sample rate values.

Not sure what's causing the 11GB of writing, however. 

Also, which version did this occur? Thanks!

**Comment by chaeyoungdul** [comment]
Hey,

I saw the metrics in activity monitor. It was in the disk tap so it would be writing on the SSD, i hope. The amount is overtime so it's added up. 

I think it would be kinda weird for it to read like 400MB for a song (mini album consisting of 2-3 songs) that's quite a lot more than the songs it self. Could you explain that?

The versions i used were the stable and beta. Both had the writing issue. I will add pictures later if they are helpful.

At least thank you for making such an app, I really appreciate it and hope to help a hand :)

Greetings,
Chaeyoung

**Comment by chaeyoungdul** [comment]
to add to the problem when I stop the music it start writing.

**Comment by vincentneo** [comment]
> I think it would be kinda weird for it to read like 400MB for a song (mini album consisting of 2-3 songs) that's quite a lot more than the songs it self. Could you explain that?

I personally don't think the logs would take up that much, so will be debugging that when I can.

> to add to the problem when I stop the music it start writing.

This is so weird.

**Comment by chaeyoungdul** [comment]
> > I think it would be kinda weird for it to read like 400MB for a song (mini album consisting of 2-3 songs) that's quite a lot more than the songs it self. Could you explain that?
> 
> I personally don't think the logs would take up that much, so will be debugging that when I can.
> 
> > to add to the problem when I stop the music it start writing.
> 
> This is so weird.

yeah idk why it does write to the disk but yeah
it writes with 10MB a second. that's very interesting.

I am on apple silicon m1 and ventura public beta 2, maybe they play a role.

**Comment by vincentneo** [comment]
<img width="931" alt="Screenshot 2022-08-04 at 6 40 49 PM" src="https://user-images.githubusercontent.com/23420208/182827892-df6e27ce-b205-4598-ad48-78aa2a7538ae.png">
Very interesting... nothings playing right now and ^. Intel mac user here on Monterey so not a specific platform issue for sure.

**Comment by vincentneo** [comment]
<img width="1592" alt="Screenshot 2022-08-04 at 6 46 44 PM" src="https://user-images.githubusercontent.com/23420208/182828858-5cb8013c-ba93-4721-a6ae-7bd1afdeaed3.png">

So my initial guess seems correct. Instruments is reporting that the read/writes came from calling `OSLogStore` APIs, which forms the foundation of how LosslessSwitcher works, so this might not be fixable.

Why would reading using `OSLogStore` write so much though? I have no idea so far...

**Comment by chaeyoungdul** [comment]
oh very interesting to see 

the reading Is not my top concern but the writing is, as it is damaging to the ssd and seems unnecessary.

**Comment by danielrdotcom** [comment]
Yep, just started using this and activity monitor does report alot of activity on the SSD which is of some concern

**Comment by DarinFlippo** [comment]
Possible that you are writing to virtual memory?

**Comment by chaeyoungdul** [comment]
> Possible that you are writing to virtual memory?

Very likely actually, though i would think that would be handled by the system  of some sort.

I sometimes do have up to 30 gigabytes in swap

**Comment by vincentneo** [comment]
> Possible that you are writing to virtual memory?

Unlikely. My (probably wrong) guess is the `OSLogStore` API does some weird processing whereby it reads the logs, write it somewhere temporarily (probably in some usable format), etc. 
We need an Apple Engineer to answer that question unfortunately.

> Very likely actually, though i would think that would be handled by the system of some sort.
> 
> I sometimes do have up to 30 gigabytes in swap

Now I'm worried about the part where you have 30GB as swap memory. What are the memory/RAM usages for LosslessSwitcher on your system? if its 30GB of memory use (aside from disk read writes), then its definitely a separate memory leak issue.

**Comment by chaeyoungdul** [comment]
> Now I'm worried about the part where you have 30GB as swap memory. What are the memory/RAM usages for LosslessSwitcher on your system? if its 30GB of memory use (aside from disk read writes), then its definitely a separate memory leak issue.
> 
> 
> 
> 

Well first of all i have only 8 gigabytes of ram. 
Second of all i use a lot of apps simultaneously logic pro is a big ram eater. 
Third of all i have a vm running at the same time. (Yes my laptop lags like crazy) grand MA 2 on pc is not for mac and so isn't MA3D which i do need

so yeah it's not a memory leak it my laptop not being adequate (the only way to get a new one is my dad.he doesn't see me needing a new laptop, but that's a separate issue).

**Comment by vincentneo** [comment]
Thanks for the clarification @chaeyoungdul
